### DISCUSSION TITLE
Querylogs are not written to files in real time?

### DISCUSSION THREAD

**Discussion by lyc0221** [AdguardTeam/AdGuardHome#3755] [Q&A]
Have a question or an idea? Please search it [on our forum](https://github.com/AdguardTeam/AdGuardHome/discussions) to make sure it was not yet asked. If you cannot find what you had in mind, please [submit it here](https://github.com/AdguardTeam/AdGuardHome/discussions/new).

### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- I am running the latest version

### Issue Details

* **Version of AdGuard Home server:**
  * The latest version
* **How did you install AdGuard Home:**
  * Docker
* **How did you setup DNS configuration:**
  * System
* **If it's a router or IoT, please write device model:**
  * 
* **CPU architecture:**
  * 
* **Operating system and version:**
  * Ubuntu 20.04.1

### Screenshots

<details><summary>Screenshot:</summary>

</details>

### Additional Information

Querylogs are not written to files in real time, how to do?

**Comment by ainar-g** [comment]
Hello.  Yes, AGH maintains an in-memory buffer of logs.  The size of that buffer is controlled by the `dns.querylog_size_memory` setting.  The lower this value is, the more often it will flush to disk, but that will affect performance.

**Reply by lyc0221** [reply]
ok, I set `dns.querylog_size_memory` to 0

**Comment by andylau004** [comment]
> ok, I set `dns.querylog_size_memory` to 0

cool.
